subprojects/gdk_rust/gdk_common/src/model.rs

```diff
@@ -0,0 +1,338 @@
+    pub num_confs: Option<usize>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct TransactionMeta {
+    #[serde(flatten)]
+    pub create_transaction: Option<CreateTransaction>,
+    #[serde(rename = "transaction")]
+    pub hex: String,
+    pub txid: String,
+    pub height: Option<u32>,
+    pub timestamp: u32, // for confirmed tx is block time for unconfirmed is when created or when list_tx happens
+    pub created_at: String, // yyyy-MM-dd HH:mm:ss of timestamp
+    pub error: String,
+    pub addressees_have_assets: bool,
+    pub is_sweep: bool,
+    pub satoshi: Balances,
+    pub fee: u64,
+    pub network: Option<Network>,
+    #[serde(rename = "type")]
+    pub type_: String, // incoming or outgoing
+    pub changes_used: Option<u32>,
+    pub rbf_optin: bool,
+    pub user_signed: bool,
+    pub spv_verified: bool,
+}
+
+impl From<BETransaction> for TransactionMeta {
+    fn from(transaction: BETransaction) -> Self {
+        let txid = transaction.txid().to_string();
+        let hex = hex::encode(&transaction.serialize());
+        let timestamp = now();
+        let rbf_optin = transaction.rbf_optin();
+
+        TransactionMeta {
+            create_transaction: None,
+            height: None,
+            created_at: format(timestamp),
+            timestamp,
+            txid,
+            hex,
+            error: "".to_string(),
+            addressees_have_assets: false,
+            is_sweep: false,
+            satoshi: HashMap::new(),
+            fee: 0,
+            network: None,
+            type_: "unknown".to_string(),
+            changes_used: None,
+            user_signed: false,
+            spv_verified: false,
+            rbf_optin,
+        }
+    }
+}
+
+impl TransactionMeta {
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        transaction: BETransaction,
+        height: Option<u32>,
+        timestamp: Option<u32>,
+        satoshi: Balances,
+        fee: u64,
+        network: Network,
+        type_: String,
+        create_transaction: CreateTransaction,
+        user_signed: bool,
+        spv_verified: bool,
+    ) -> Self {
+        let mut wgtx: TransactionMeta = transaction.into();
+        let timestamp = timestamp.unwrap_or_else(now);
+        let created_at = format(timestamp);
+
+        wgtx.create_transaction = Some(create_transaction);
+        wgtx.height = height;
+        wgtx.timestamp = timestamp;
+        wgtx.created_at = created_at;
+        wgtx.satoshi = satoshi;
+        wgtx.network = Some(network);
+        wgtx.fee = fee;
+        wgtx.type_ = type_;
+        wgtx.user_signed = user_signed;
+        wgtx.spv_verified = spv_verified;
+        wgtx
+    }
+}
+
+#[derive(Debug)]
+pub struct AddressIO {
+    pub address: String,
+    pub address_type: bitcoin::util::address::AddressType,
+    pub addressee: String,
+    pub is_output: String,
+    pub is_relevant: String,
+    pub is_spent: String,
+    pub pointer: u32,
```

Comment:
> I had to look up the usage in the code to understand what this does. Maybe worth adding a comment, or possibly even renaming to child_number as used in rust-bitcoin and bip32?

I do agree `child_number` would be a better fit but pointer is part of the gdk interface https://gdk.readthedocs.io/en/release_0.0.29/gdk-json.html#receive-address-json (and in the future, I want to remove functions like `address_io_value` in the `serialize.rs` and let serde do its magic automatically. While you can do the rename when serializing (keeping `child_number` in the struct while serializing `pointer` in JSON) I am not sure it's more clear overall

> Also, does this struct specify whether the address was derived from the internal or external chains?

I think for gdk interface it's implicitly  an external one, (since changes are never returned as JSON?)

Reply:
Alright, agreed, if this is the lingo in GDK then it make sense to be consistent. I would just add a short comment in the struct definition explaining that `pointer` is the child derivation index.

Reply:
added in 943f85c7e2d5cb34d75dec9604a7d1ce7bcacc5e